src/containers/SearchPage/SearchSaveButton.tsx

```diff
@@ -80,10 +161,20 @@ const SearchSaveButton = ({ userData }: Props) => {
       handleFailure("fetchFailed");
       return;
     }
-
     const newSearch = createSearchString(window.location);
-    const newSearchList = [getSavedSearchRelativeUrl(newSearch), ...oldSearchList];
-    if (!oldSearchList.find((s) => s === getSavedSearchRelativeUrl(newSearch))) {
+
+    // Need to remove query if it is not in search url as it can exist in search input only
+    const actualSelectors = location.search.includes("query")
+      ? selectors
+      : selectors.map((s) => (s.parameterName === "query" ? { ...s, value: "" } : s));
```

Comment:
Dette er selvsagt smak og behag. Jeg følger både `s` og `selector` her, og jeg veit ikke i farta hva jeg hadde skrevet sjøl (jeg synder på 1-bokstav-variabler selv fra tid til annen).

```suggestion
      : selectors.map((selector) => (selector.parameterName === "query" ? { ...selector, value: "" } : selector));
```